### ISSUE TITLE
Youtube API change? Some channels have /user/{username}, but some have /c/{username}

### ISSUE THREAD

**Issue by dscottboggs** [flaviotordini/minitube#218]
I just installed minitube, and began searching for my typical subscriptions to add them to it. When I got to [Ben Eater](https://www.youtube.com/c/BenEater), clicking the username in the search results led me instead to the channel for one [Bill E. Neater](https://www.youtube.com/user/beneater) (titled "William Neater").

I think this is *probably* related to #207, but I haven't looked at the code so I'm not sure. What does seem clear is that minitube is expecting all channels to be at `youtube.com/user/{user id}`, but youtube now has an alternate `youtube.com/c/{user id}` endpoint as well.

**Comment by PaperNick** [comment]
Stumbled upon the same issue. Looking at the source code and how channel subscriptions are stored in the SQLite database, it is indeed relying on the `user_id`.

Moving forward, I think the app should operate only with the channel id.

**Comment by dscottboggs** [comment]
Where do you see that? I just went looking for `user_id` and found [this](https://github.com/flaviotordini/minitube/blob/cb3410f23f78913441fd4929772c441bf089e8d5/src/database.cpp#L103) which seems to indicate otherwise. Looking [here](https://github.com/flaviotordini/minitube/blob/cb3410f23f78913441fd4929772c441bf089e8d5/src/ytchannel.cpp#L88-L94) it looks like they're using the [correct API](https://developers.google.com/youtube/v3/docs/channels/list)? But I'm not sure.

Maybe I can run the app in a debug mode and learn why it's behaving this way if I have the time.

**Comment by PaperNick** [comment]
Yeah, my bad. I read the comment `this is really channel_id` right after I posted.

I haven't look at the code, but noticed that when I press "subscribe", the `user_id` is saved in the SQLite rather than the `channel_id` (which starts with `UCxxxxxxxx`).

Since Minitube is using the Youtube API, it might be worth to check what [youtube/v3/docs/channels/list](https://developers.google.com/youtube/v3/docs/channels/list) returns when a `user_id` is given in the `id` query param instead of `channel_id`. Maybe it falls back to `forUsername` param?
